### ISSUE TITLE
Alchemical Boilers extremely laggy

### ISSUE THREAD

**Issue by JasonMcRay** [Tuhljin/Automagy#49]
If Alchemical Boiler contains a fluid it cause fairly big tick rate.

![image](https://cloud.githubusercontent.com/assets/4236895/16227319/42853718-37b0-11e6-82f2-8d511b0789e8.png)
Above picture yes it is little overkill, but bellow you can see OPIS tick data. As you can see one boiler has tick rate of 800ms which is fairly high for single machine. The boilers have EnderIO Reservoirs underneath them to supply them with infinite water.
![image](https://cloud.githubusercontent.com/assets/4236895/16227365/6dd390ae-37b0-11e6-88be-b6ccf20d7f4e.png)

Another picture taken from opis run on InfiTech 2 Server
![image](https://cloud.githubusercontent.com/assets/4236895/16227376/81fce594-37b0-11e6-9b4c-f6443cefca64.png)
4th Line has only 5 Boilers but total tick time of them is above 1300ms

Currently the only workaround is to pump the fluid out of them if they are not being used, or needed.

Automagy: v0.28.2 for MC 1.7.10
Modpack: InfiTech 2

**Comment by Tuhljin** [comment]
Let's try to determine where the slowdown actually comes from. If the boiler has water in it, but there isn't something actively trying to pump more water into it, is there still an issue? How does this compare to an alchemical furnace that's constantly being fed items and fuel? Or are the boilers not even being fueled? Are there no alembics at work? It doesn't look like it, from that screenshot.

**Comment by JasonMcRay** [comment]
Same setup as in the picture above (same number of the Boilers = 48)

Base tick rate without any fuel/water in them: 0.884 µs - 0.018 µs per boiler

Without fuel in boilers (idle)
1) Boilers in progress of getting filled - 13393.121 µs - 279.023 µs per boiler
2) Boilers have fluid but not actively pumping in - 13258.008 µs - 276.209 µs per boiler
3) Boilers full on fluid and still trying to push fluid in - 13342.229 µs - 277.963 µs per boiler

With fuel in boiler (working)
all 3 tests roughly the same results: 13854.242 µs - 288.630 µs per boiler

And no Alembics in place

I am personally unable to recreate the massive 30000µs from what I got from one of my players, but still it seems pretty high per boiler.

Also did test with only 1 boiler filled with water idling = 1065.310 µs
2 boilers = 1526.303 µs - 763.151 µs per boiler
4 boilers = 2026.192 µs - 506.548 µs per boiler

Just as a note. I have done a little test with only EnderIO , Thaumcraft and Automagy (so outside of the modpack). The tick rate in those tests was around 1462 - 30 per boiler. So I guess there is some incompatibility somewhere. But disabling mod by mod (with over 100 mods installed) I don't think I will have strength and time to do that.

**Comment by Castone22** [comment]
I don't think this is necessarily an issue with the alchemical boiler, but more with the way that they and the alchemical furnace interface with alembics.  From what i've caught using warmroast to inspect the thread it seems like every tick the furnace asks itself if there is room in its essentia tank for something to burn.  

It will also do a check occasionally to determine if any attached alembic has room for one of the essentia types in its tank, and if it does, deposit to the first alembic that has room for that essentia type.

I'm not certain there exists a great way to fix this without rewriting the way TC handles basic alchemy burners.
